src/components/tl/mlx5/mcast/tl_mlx5_mcast_team.c

```diff
@@ -88,11 +88,15 @@ ucc_status_t ucc_tl_mlx5_mcast_team_init(ucc_base_context_t *base_context,
 
     memcpy(&comm->params, conf_params, sizeof(*conf_params));
 
-    comm->wsize     = conf_params->wsize;
-    comm->max_eager = conf_params->max_eager;
-    comm->comm_id   = team_params->id;
-    comm->ctx       = mcast_context;
-    comm->grh_buf   = (char *)ucc_malloc(GRH_LENGTH * sizeof(char), "grh_buf");
+    comm->one_sided.reliability_enabled     = conf_params->one_sided_reliability_enable;
+    comm->wsize                             = conf_params->wsize;
+    comm->max_push_send                     = conf_params->max_push_send;
+    comm->max_eager                         = conf_params->max_eager;
+    comm->comm_id                           = team_params->id;
+    comm->ctx                               = mcast_context;
+    comm->mcast_group_count                 = 1; /* TODO: add support for more number of mcast groups */
```

Comment:
if this is always 1 why is it useful to have this field?

Reply:
we will have the design for multiple mcast groups in the future PRs. Once the PRs are in, this value will be read from an env var that user provides.

Reply:
I would suggest to add it in the relevant pr then

Reply:
the thing is that ```mcast_group_count``` is getting read in other functions and we need it there. it is just that since ```mcast_group_count``` is 1, then index of QP will become always 0. But we still can handle more number of mcast groups.    what I meant by _**we will have the design for multiple mcast groups in the future PRs**_ is that we will add the connection initialization related codes (init codes) of multiple mcast groups in the future.